nifi-nar-bundles/nifi-graph-bundle/nifi-other-graph-services/src/main/java/org/apache/nifi/graph/TinkerpopClientService.java

```diff
@@ -0,0 +1,538 @@
+            .dependsOn(CONNECTION_SETTINGS, SERVICE_SETTINGS)
+            .build();
+
+    public static final PropertyDescriptor PATH = new PropertyDescriptor.Builder()
+            .name("tinkerpop-path")
+            .displayName("Path")
+            .description("The URL path where Gremlin Server is running on each host listed as a contact point.")
+            .required(true)
+            .defaultValue("/gremlin")
+            .addValidator(StandardValidators.NON_EMPTY_EL_VALIDATOR)
+            .expressionLanguageSupported(ExpressionLanguageScope.VARIABLE_REGISTRY)
+            .dependsOn(CONNECTION_SETTINGS, SERVICE_SETTINGS)
+            .build();
+
+    public static final PropertyDescriptor TRAVERSAL_SOURCE_NAME = new PropertyDescriptor.Builder()
+            .name("gremlin-traversal-source-name")
+            .displayName("Traversal Source Name")
+            .description("An optional property that lets you set the name of the remote traversal instance. " +
+                    "This can be really important when working with databases like JanusGraph that support " +
+                    "multiple backend traversal configurations simultaneously.")
+            .expressionLanguageSupported(ExpressionLanguageScope.VARIABLE_REGISTRY)
+            .addValidator(Validator.VALID)
+            .build();
+
+    public static final PropertyDescriptor REMOTE_OBJECTS_FILE = new PropertyDescriptor.Builder()
+            .name("remote-objects-file")
+            .displayName("Remote Objects File")
+            .description("The remote-objects file yaml used for connecting to the gremlin server. Only the yaml file or the string can be specified.")
+            .required(true)
+            .addValidator(StandardValidators.FILE_EXISTS_VALIDATOR)
+            .expressionLanguageSupported(ExpressionLanguageScope.VARIABLE_REGISTRY)
+            .dependsOn(CONNECTION_SETTINGS, YAML_SETTINGS)
+            .build();
+
+    public static final PropertyDescriptor USER_NAME = new PropertyDescriptor.Builder()
+            .name("user-name")
+            .displayName("Username")
+            .description("The username used to authenticate with the gremlin server." +
+                    " Note: when using a remote.yaml file, this username (if set) will overload any username set in the yaml file.")
+            .required(false)
+            .expressionLanguageSupported(ExpressionLanguageScope.VARIABLE_REGISTRY)
+            .build();
+
+    public static final PropertyDescriptor PASSWORD = new PropertyDescriptor.Builder()
+            .name("password")
+            .displayName("Password")
+            .description("The password used to authenticate with the gremlin server." +
+                    " Note: when using a remote.yaml file, this password (if set) will override any password set in the yaml file")
+            .required(false)
+            .expressionLanguageSupported(ExpressionLanguageScope.VARIABLE_REGISTRY)
```

Comment:
Sensitive properties should not support expression language, instead, they can be supplied through Parameter Contexts if necessary.
```suggestion
```

Reply:
Removed.